### PR TITLE
Improve sort

### LINKED ISSUE #64: Improve Sort Command
Unify attributes that can be sorted under an abstract class Attribute with a method getAttributeComparator() that can return a Comparator.

### LINKED ISSUE #66: "sort asc" causes crash

### PR DESCRIPTION
**Pull request by highorbit25**
Logic for sorting attributes are now consolidated under the abstract class `Attribute`
- Static `isValidAttributeType(String test)` method verifies valid input attribute by checking against static  `ATTRIBUTE_SET` 
- `getAttributeComparator(String attributeType)` returns the Comparator to sort the given attribute
- To sort new attributes in the future, we just need to add the attribute into `ATTRIBUTE_SET` and specify the Comparator in this abstract class.
- Private constructor in 'Order' class so that only the same static `ORDER_ASC` & `ORDER_DSC` is used as the sorting order.
- Fix #66 
- Resolves #64

### CODE REVIEW

**Comment by highorbit25** [comment]
Did not implement Attribute as a super class for attributes to inherit as per suggested in previous PR discussion.
Felt that having each attribute that extends Attribute overwrite getAttributeComparator() individually would not value add much. Having all the Comparator and valid attributeTypes located within the abstract Attribute class itself for ease of future development seems sufficient for now.
Open for further discussion!

File: src/main/java/seedu/studmap/model/attribute/Attribute.java [hunk 1]
```diff
@@ -1,58 +1,60 @@
...
-     * Format state as text for viewing.
+     * Returns the Comparator to sort the given attribute
+     *
      */
-    public String toString() {
-        return '[' + attributeName + ']';
+    public static Comparator<Student> getAttributeComparator(String attributeType) throws AttributeNotFoundException {
+        Comparator<Student> resultComparator = Comparator.comparing(Student::getNameString);
+        if (isValidAttributeType(attributeType)) {
+            switch (attributeType) {
```
Comment (Piyotato) [inline src/main/java/seedu/studmap/model/attribute/Attribute.java:32]: Should use `enum` for `AttributeType`. Use `valueOf` to parse the input in the parser. See: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Enum.html#valueOf(java.lang.Class,java.lang.String)

**Review by Piyotato — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/studmap/model/attribute/Attribute.java (hunk 1 above)
Comment (highorbit25) [reply src/main/java/seedu/studmap/model/attribute/Attribute.java]: Agreed, implemented accordingly. Thanks!

**Review by Piyotato — APPROVED** [review]
LGTM

File: src/main/java/seedu/studmap/model/attribute/AttributeType.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package seedu.studmap.model.attribute;
+
+/**
+ * Enum class for valid attributes supported by sorting.
+ */
+public enum AttributeType {
```
Comment (lpohsien) [inline src/main/java/seedu/studmap/model/attribute/AttributeType.java:6]: Just a thought, will it be better (in terms of organization and readability) if AttributeType is an inner class of Attribute? Or is there any particular reason requires the AttributeType to be separated?

**Review by lpohsien — APPROVED** [review]
Good use of OOP to make sorting extensible. LGTM.

**Pull request merged**